cocotb/handle.py

```diff
@@ -752,7 +754,10 @@ def setimmediatevalue(self, value):
             TypeError: If target has an unsupported type for
                  string value assignment.
         """
-        if not isinstance(value, str):
+        if isinstance(value, str):
```

Comment:
This is just for convenience. It was likely this was the original intent, and will produce values safe for the simulator to consume or will throw an exception.

Reply:
Perhaps add a `DeprecationWarning` on this leg.